xonsh/history/base.py

```diff
@@ -0,0 +1,99 @@
+import threading
+import uuid
+
+import xonsh.tools as xt
+
+
+class HistoryGC(threading.Thread):
+    pass
+
+
+class HistoryBase:
+    def __init__(self, sessionid=None, gc=True, **kwargs):
+        self.sessionid = uuid.uuid4() if sessionid is None else sessionid
+        self.gc = HistoryGC() if gc else None
```

Comment:
HistoryGC doesn't seem to do anything, so maybe gc should just be set to None on the base?